pkg/apis/kubeone/types.go

```diff
@@ -52,6 +52,23 @@ type KubeOneCluster struct {
 	Credentials map[string]string `json:"credentials,omitempty"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeOneSecrets hold secrets needed to provision the cluster
+type KubeOneSecrets struct {
```

Comment:
This adds a new API type. Eventually, we don't have to use API type like this, but have some package like the old API (`pkg/config`), although I prefer this way.